lib/services/ios-debug-service.ts

```diff
@@ -167,13 +176,17 @@ class IOSDebugService implements IDebugService {
 		return (() => {
 			let timeout = this.getReadyForAttachTimeout();
 			this.$iOSSocketRequestExecutor.executeAttachRequest(device, timeout).wait();
-			this.wireDebuggerClient(() => device.connectToPort(inspectorBackendPort)).wait();
+			this.wireDebuggerClient(device).wait();
 		}).future<void>()();
 	}
 
-	private wireDebuggerClient(factory: () => net.Socket): IFuture<void> {
+	private wireDebuggerClient(device: Mobile.IiOSDevice): IFuture<void> {
```

Comment:
You can make the device argument not mandatory
```TypeScript
private wireDebuggerClient(device?: Mobile.IiOSDevice): IFuture<void> {
```

This way you will not have to pass `null` when calling the method without device.